src/crate/client/sqlalchemy/tests/insert_from_select_test.py

```diff
@@ -66,17 +68,41 @@ class CharacterArchive(Base):
         self.character_archived = CharacterArchive
         self.session = Session()
 
+    @unittest.skipIf(SA_VERSION >= SA_1_4, "Not supported on SQLAlchemy >= 1.4")
     @patch('crate.client.connection.Cursor', FakeCursor)
-    def test_insert_from_select_triggered(self):
+    def test_insert_from_select_triggered_1_0_0(self):
+
+        self.engine.dialect.server_version_info = (1, 0, 0)
+
         char = self.character(name='Arthur', status='Archived')
         self.session.add(char)
         self.session.commit()
 
         sel = select([self.character.name, self.character.age]).where(self.character.status == "Archived")
         ins = insert(self.character_archived).from_select(['name', 'age'], sel)
+
         self.session.execute(ins)
         self.session.commit()
         self.assertSQL(
             "INSERT INTO characters_archive (name, age) (SELECT characters.name, characters.age FROM characters WHERE characters.status = ?)",
             ins
         )
+
+    @patch('crate.client.connection.Cursor', FakeCursor)
+    def test_insert_from_select_triggered_1_0_1(self):
```

Comment:
Is this test needed? Even on recent CrateDB versions, adding parenthesis wouldn't hurt. Look like this tests sqlalchemy and not crate specific code...

Reply:
The tests have already been there. They seem to specifically, and intentionally, test a change in the CrateCompiler's behavior when observing to be connected to CrateDB >= 1.0.1.

In fact, not using parentheses here is the standard behavior of SQLAlchemy. So, if we consider to drop support for CrateDB < 1.0.1, we could get rid of having to vendor and override the dialect's `visit_insert` method at all.

https://github.com/crate/crate-python/blob/02f6529d98ad75c1bb6f43f5e4ac3f81003e7f93/src/crate/client/sqlalchemy/compiler.py#L204-L219

In turn, our test suite would implictly increase code coverage because of the reduced surface of vendored code, see also https://github.com/crate/crate-python/pull/391#pullrequestreview-981282894.